edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/handlers/SubscriptionChangeHandler.cs

```diff
@@ -86,8 +91,86 @@ public async Task<bool> HandleAsync(MqttPublishInfo publishInfo)
                 return true;
             }
 
-            // TODO: the following solution is to unblock the case when in a nested scenario a child edgehub
-            // subscribes in the name of a device/module. However changes needed to support unsubscribe
+            var id1 = match.Groups["id1"];
+            var id2 = match.Groups["id2"];
+
+            // For indirect clients the subscription change will be reported through edgeHub. That is a
+            // slower method to reconcile so we handle the direct case separately with a faster method
+            if (id2.Success && string.Equals(id2.Value, Constants.EdgeHubModuleId))
+            {
+                await this.HandleIndirectChanges(subscriptionList);
+            }
+            else
+            {
+                await this.HandleDirectChanges(id1, id2, subscriptionList);
+            }
+
+            return true;
+        }
+
+        async Task HandleDirectChanges(Group id1, Group id2, List<string> subscriptionList)
```

Comment:
This id1 and id2 makes this hard to follow. Can we expand these before we pass them around..